awx/api/serializers.py

```diff
@@ -3894,6 +3894,26 @@ def validate(self, attrs):
             raise serializers.ValidationError(_('Must include "username" and "password".'))
 
 
+class UserSessionSerializer(serializers.Serializer):
+
+    username = serializers.CharField()
+    password = serializers.CharField()
+
+    def validate(self, attrs):
+        username = attrs.get('username')
+        password = attrs.get('password')
+
+        if username and password:
+            user = authenticate(username=username, password=password)
```

Comment:
I *really* don't want authentication happening here. serializer based validation is really meant to validate inputs and structure... not to authenticate.

Reply:
For instance, this is really the method you would use when creating a user.

Reply:
This one follows the convention of `AuthTokenSerializer`, which did put authentication inside `validate`. But it makes sense to distinguish between 400 and 401 errors.

Reply:
We definitely need to just rely on the drf views and serializers. There is no need to do this but ourselves.